server/src/main/java/keywhiz/service/daos/SecretContentDAO.java

```diff
@@ -74,12 +78,39 @@ public long createSecretContent(long secretId, String encryptedContent,
     r.setExpiry(expiry);
     r.store();
 
+    pruneOldContents(secretId);
```

Comment:
I would prefer to do last N instead of last N days. Otherwise there are two risks: we might not keep the previous secret (if it was older than 45 days) or we might keep too many old secrets.

We can use the auto increment ID to figure out the order in which the content was created.